packages/form-builder/src/blocks/fields/amount/Edit.tsx

```diff
@@ -1,51 +1,201 @@
+            ? periodLookup[recurringBillingPeriod].plural
+            : periodLookup[recurringBillingPeriodOptions[0]].plural;
+
+        return (
+            <strong>
+                {1 === interval && <>{label ?? singular}</>}
+                {1 !== interval && (
+                    <>
+                        {interval} {label ?? plural}
+                    </>
+                )}
+            </strong>
+        );
+    };
 
-    const FixedPriceMessage = () => {
-        const amount = formatCurrencyAmount(setPrice);
+    const RecurringMessage = () => {
+        const interval = parseInt(recurringBillingInterval);
 
-        return createInterpolateElement(__('This donation is set to <amount/> for this form.', 'give'), {
-            amount: <strong>{amount}</strong>,
-        });
+        const periodPlaceholder = <strong>{'[' + _n('period', 'periods', interval, 'give') + ']'}</strong>;
+
+        const periodElement = !isRecurringDonorChoice ? (
+            <RecurringPeriod count={interval} label={periodPlaceholder} />
+        ) : (
+            <RecurringPeriod count={interval} label={periodPlaceholder} />
+        );
+
+        const count = parseInt(recurringLengthOfTime);
+        const countElement = <strong>{count}</strong>;
+
+        const translatableString = !count
+            ? __('This donation occurs every <period />.', 'give')
+            : __('This donation occurs every <period /> for <count /> <strong>payments</strong>.', 'give');
+
+        return (
+            <Notice>
+                {createInterpolateElement(translatableString, {
+                    period: periodElement,
+                    count: countElement,
+                    strong: <strong />, // Required to interpolate the strong tag around the "payments" text.
+                })}
+            </Notice>
+        );
+    };
+
+    const FixedRecurringMessage = () => (
+        <Notice>
+            {createInterpolateElement(__('This donation is <amount/> every <period/>. ', 'give'), {
+                amount: <strong>{amountFormatted}</strong>,
+                period: <RecurringPeriod count={parseInt(recurringBillingInterval)} label={undefined} />,
+            })}
+        </Notice>
+    );
+
+    const BillingPeriodControl = ({options}) => {
+        const [option, setOption] = useState(options[0]);
+        return (
+            <RadioControl
+                className={'give-billing-period-control'}
+                label={__('Billing Period', 'give')}
+                hideLabelFromVision={true}
+                selected={option}
+                options={['one-time'].concat(options).map((option) => {
+                    return {
+                        label: 'one-time' === option ? __('One Time', 'give') : periodLookup[option].adjective,
+                        value: option,
+                    };
+                })}
+                onChange={(value) => setOption(value)}
+                //@ts-ignore
```

Comment:
typescript error, I don't think disabled is a valid prop here

Reply:
This can probably just be removed.

Reply:
Resolved, see https://github.com/impress-org/givewp-next-gen/pull/128/commits/b53299ed159c1743bd2b48a429fb4606c54001e5.